datamade/code-challenge
# Rowen Trombo's Submission of Code Challenge

## Overview

Completed the steps outlined in the README file and adjusted as necessary for testing passes. Outside of the steps in README changes included:
-Minor updates to install newest versions in Dockerfile and .ymp files
-Update to API url in 'templates/urls.py' to match that url given in 'templates/parserator_web/index.html'



### Demo
Updating for 3.8 import in Dockerfile
![image](https://github.com/user-attachments/assets/3d6ac78f-7f5b-44e6-851c-a67472dd447e)
![image](https://github.com/user-attachments/assets/8c99a84b-2c55-432b-8b07-921dd184eba5)


in 'templates/parserator_web/index.html'
![image](https://github.com/user-attachments/assets/f2271c3b-f370-4bb2-9124-a3b6d4b8ce52)

in 'templates/urls.py' 
![image](https://github.com/user-attachments/assets/41016f45-9435-4fbb-8d6b-0df749d89f89)


### Notes

I used heavier commenting than may be expected for best practice to be as explantory as possible with my edits.

## Testing Instructions

To start testing create the Docker build in Command Prompt:
```
docker-compose build
```

Next, run the app:

```
docker-compose up
```
You can run the tests using Docker:

```bash
docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app
```
The test results should look as so:
![image](https://github.com/user-attachments/assets/d8a65c6b-e679-44e8-8774-a3da7ef653cc)

*Please note: the XFAIL errors are referring to expected to fail errors*
*Also note the [NOTRUN] tests are referring to tests not run due to ```yield``` not being supported by the current version of ```pytest```*
